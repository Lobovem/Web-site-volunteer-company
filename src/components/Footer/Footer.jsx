import { FooterMenu } from '../FooterMenu/FooterMenu';
import s from './Footer.module.scss';

import { FooterContacts } from '../FooterContacts/FooterContacts';
import { FooterSocial } from '../FooterSocial/FooterSocial';
import { Btn } from '../kit/Btn/Btn';
import { Link } from 'react-router-dom';

export const Footer = () => {
  return (
    <section className={s.footer}>
      <img className={s.footer__img} src="./img/logo-second.png" alt="logo-second.png" />
      <div className={s.footer__wrap}>
        <div className={s.footer__inner}>
          <FooterMenu />
          <FooterContacts />
          <FooterSocial />

          <div className={s.footer__descBox}>
            <p className={s.footer__desc}>
              Україна – це вічність, не тільки сьогоднішня, але передусім майбутня й минула. Самостійну державу може здобути собі
              український народ тільки власною боротьбою і трудом. Не дивіться на Україну, як на землю своїх батьків. Дивіться на неї, як на
              землю своїх дітей
            </p>
            <div className={s.footer__btnBox}>
              <Link to="/donate">
                <Btn title="задонатити" type="button" className="btn-secondary" />
              </Link>
            </div>
          </div>
        </div>

        <div className={s.footer__copyright}>©‌ Міжнародний благодійний фонд “Український гуманітарний батальйон”, 2022-2023</div>
      </div>
    </section>
  );
};
