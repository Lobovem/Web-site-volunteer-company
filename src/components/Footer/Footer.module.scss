@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.footer {
  position: relative;
  background: linear-gradient(180deg, #025FA7 0%, #1682D4 53.31%, #00B0D0 87.89%, #09D3F6 124.3%, #42E3FF 153.83%);
  border-radius: 0px;

  &__img {
    max-width: 133px;
    position: absolute;
    top: -60px;
    left: 50%;
    right: 50%;
    transform: translate(-50%, 0);

    @include lg-min {
      max-width: 224px;
      top: -100px;

    }
  }

  &__wrap {
    @include layoutMaxPadding;
    padding: 150px 0px 90px 00px;

  }

  &__inner {
    display: flex;
    flex-direction: column;
    gap: 40px;
    justify-content: space-between;

    @media(min-width: 700px) {
      flex-wrap: wrap;
      flex-direction: row;
    }

    @include lg-min {
      flex-wrap: nowrap;
    }
  }

  &__descBox {
    @include lg-min {
      flex: 1;
    }
  }

  &__desc {



    font-weight: $font-weight-400;
    font-size: $font-size-14;
    line-height: 20px;
    color: $color-white;
    mix-blend-mode: normal;
    padding-bottom: 40px;
    margin: 0;

  }

  &__btnBox {
    display: flex;
    justify-content: center;
  }

  &__copyright {



    border-top: 2px solid $color-white;
    padding-top: 40px;
    font-weight: $font-weight-500;
    font-size: $font-size-14;
    line-height: 17px;
    color: $color-white;
    mix-blend-mode: normal;
    margin-top: 70px;
    width: 100%;

    // text-align: center;
    @include xs-min {
      text-align: center;
    }
  }
}