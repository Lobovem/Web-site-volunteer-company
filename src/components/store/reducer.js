import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
