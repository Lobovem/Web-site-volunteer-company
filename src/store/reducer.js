import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
