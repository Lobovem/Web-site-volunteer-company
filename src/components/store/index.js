// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
