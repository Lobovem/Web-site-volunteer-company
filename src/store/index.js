// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
