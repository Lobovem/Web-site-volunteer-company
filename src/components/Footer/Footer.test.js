import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Footer } from './Footer';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('Footer component', () => {
  test('renders Footer component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the title component menu', () => {
    render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /ПРО ФОНД/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('should render the title component contacts', () => {
    render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /КОНТАКТИ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('should render the title component social', () => {
    render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    const titleSection = screen.getByRole('heading', { name: /СОЦІАЛЬНІ МЕРЕЖІ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('renders and search component button', () => {
    render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    const buttonElement = screen.getByRole('button', { type: 'button' });
    expect(buttonElement).toBeInTheDocument();
  });

  test('renders Footer component with 7 images (pay elements)', () => {
    render(
      <Provider store={store}>
        <Router>
          <Footer />
        </Router>
      </Provider>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(5);
  });
});
