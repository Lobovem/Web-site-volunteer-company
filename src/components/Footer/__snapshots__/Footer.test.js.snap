// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
      >
        <img
          alt="logo-second.png"
          class="footer__img"
          src="/img/logo-second.png"
        />
        <div
          class="footer__wrap"
        >
          <div
            class="footer__inner"
          >
            <nav
              class="menu"
            >
              <div
                class="menu__wrap"
              >
                <h2
                  class="menu__title"
                >
                  про фонд
                </h2>
                <ul
                  class="menu__list"
                />
              </div>
            </nav>
            <div
              class="contacts"
            >
              <div
                class="contacts__wrap"
              >
                <h2
                  class="contacts__title"
                >
                  контакти
                </h2>
                <div
                  class="contacts__phoneBox"
                >
                  <img
                    alt="icon-phone.svg"
                    class="contacts__img"
                    src="/img/icon-phone.svg"
                  />
                  <a
                    class="contacts__link"
                    href="tel:+380630494002"
                  >
                    +380 (63)-049-40-02
                  </a>
                  <a
                    class="contacts__link"
                    href="tel:+380671597373"
                  >
                    +380 (67)-159-73-73
                  </a>
                </div>
                <div
                  class="contacts__adressBox"
                >
                  <img
                    alt="icon-adress.svg"
                    class="contacts__img"
                    src="/img/icon-adress.svg"
                  />
                  <a
                    class="contacts__link"
                    href="https://goo.gl/maps/uXTuc6J6nSarrwsW7"
                    target="blank"
                  >
                    <address
                      class="contacts__desc"
                    >
                      м. Харків, вул.Космічна, буд. 21, оф. 529
                    </address>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="social"
            >
              <div
                class="social__wrap"
              >
                <h2
                  class="social__title"
                >
                  соціальні мережі
                </h2>
                <a
                  class="social__wrapLink"
                  href="https://www.instagram.com/ukrhumbat/?igshid=YmMyMTA2M2Y="
                  target="blank"
                >
                  <img
                    alt="icon-instagram.svg"
                    class="social__img"
                    src="/img/icon-instagram.svg"
                  />
                  <p
                    class="social__desc"
                  >
                    instagram
                  </p>
                </a>
                <a
                  class="social__wrapLink"
                  href="https://www.facebook.com/people/%D0%9C%D0%91%D0%A4-%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B8%D0%B9-%D0%B3%D1%83%D0%BC%D0%B0%D0%BD%D1%96%D1%82%D0%B0%D1%80%D0%BD%D0%B8%D0%B9-%D0%B1%D0%B0%D1%82%D0%B0%D0%BB%D1%8C%D0%BE%D0%BD/100085936770532/"
                  target="blank"
                >
                  <img
                    alt="icon-facebook.svg"
                    class="social__img"
                    src="/img/icon-facebook.svg"
                  />
                  <p
                    class="social__desc"
                  >
                    facebook
                  </p>
                </a>
              </div>
            </div>
            <div
              class="footer__descBox"
            >
              <p
                class="footer__desc"
              >
                Україна – це вічність, не тільки сьогоднішня, але передусім майбутня й минула. Самостійну державу може здобути собі український народ тільки власною боротьбою і трудом. Не дивіться на Україну, як на землю своїх батьків. Дивіться на неї, як на землю своїх дітей
              </p>
              <div
                class="footer__btnBox"
              >
                <a
                  href="/donate"
                >
                  <button
                    class="btn-secondary"
                    type="button"
                  >
                    задонатити
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="footer__copyright"
          >
            ©‌ Міжнародний благодійний фонд “Український гуманітарний батальйон”, 2022-2023
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
    >
      <img
        alt="logo-second.png"
        class="footer__img"
        src="/img/logo-second.png"
      />
      <div
        class="footer__wrap"
      >
        <div
          class="footer__inner"
        >
          <nav
            class="menu"
          >
            <div
              class="menu__wrap"
            >
              <h2
                class="menu__title"
              >
                про фонд
              </h2>
              <ul
                class="menu__list"
              />
            </div>
          </nav>
          <div
            class="contacts"
          >
            <div
              class="contacts__wrap"
            >
              <h2
                class="contacts__title"
              >
                контакти
              </h2>
              <div
                class="contacts__phoneBox"
              >
                <img
                  alt="icon-phone.svg"
                  class="contacts__img"
                  src="/img/icon-phone.svg"
                />
                <a
                  class="contacts__link"
                  href="tel:+380630494002"
                >
                  +380 (63)-049-40-02
                </a>
                <a
                  class="contacts__link"
                  href="tel:+380671597373"
                >
                  +380 (67)-159-73-73
                </a>
              </div>
              <div
                class="contacts__adressBox"
              >
                <img
                  alt="icon-adress.svg"
                  class="contacts__img"
                  src="/img/icon-adress.svg"
                />
                <a
                  class="contacts__link"
                  href="https://goo.gl/maps/uXTuc6J6nSarrwsW7"
                  target="blank"
                >
                  <address
                    class="contacts__desc"
                  >
                    м. Харків, вул.Космічна, буд. 21, оф. 529
                  </address>
                </a>
              </div>
            </div>
          </div>
          <div
            class="social"
          >
            <div
              class="social__wrap"
            >
              <h2
                class="social__title"
              >
                соціальні мережі
              </h2>
              <a
                class="social__wrapLink"
                href="https://www.instagram.com/ukrhumbat/?igshid=YmMyMTA2M2Y="
                target="blank"
              >
                <img
                  alt="icon-instagram.svg"
                  class="social__img"
                  src="/img/icon-instagram.svg"
                />
                <p
                  class="social__desc"
                >
                  instagram
                </p>
              </a>
              <a
                class="social__wrapLink"
                href="https://www.facebook.com/people/%D0%9C%D0%91%D0%A4-%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B8%D0%B9-%D0%B3%D1%83%D0%BC%D0%B0%D0%BD%D1%96%D1%82%D0%B0%D1%80%D0%BD%D0%B8%D0%B9-%D0%B1%D0%B0%D1%82%D0%B0%D0%BB%D1%8C%D0%BE%D0%BD/100085936770532/"
                target="blank"
              >
                <img
                  alt="icon-facebook.svg"
                  class="social__img"
                  src="/img/icon-facebook.svg"
                />
                <p
                  class="social__desc"
                >
                  facebook
                </p>
              </a>
            </div>
          </div>
          <div
            class="footer__descBox"
          >
            <p
              class="footer__desc"
            >
              Україна – це вічність, не тільки сьогоднішня, але передусім майбутня й минула. Самостійну державу може здобути собі український народ тільки власною боротьбою і трудом. Не дивіться на Україну, як на землю своїх батьків. Дивіться на неї, як на землю своїх дітей
            </p>
            <div
              class="footer__btnBox"
            >
              <a
                href="/donate"
              >
                <button
                  class="btn-secondary"
                  type="button"
                >
                  задонатити
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="footer__copyright"
        >
          ©‌ Міжнародний благодійний фонд “Український гуманітарний батальйон”, 2022-2023
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
